src/core/server/ui_settings/settings/theme.ts

```diff
@@ -6,17 +6,45 @@
  * Side Public License, v 1.
  */
 
-import { schema } from '@kbn/config-schema';
+import { schema, Type } from '@kbn/config-schema';
 import { i18n } from '@kbn/i18n';
 import { UiSettingsParams } from '../../../types';
 
-export const getThemeSettings = (): Record<string, UiSettingsParams> => {
+function parseThemeTags() {
+  if (!process.env.KBN_OPTIMIZER_THEMES) {
+    return ['v8light', 'v8dark'];
+  }
+
+  if (process.env.KBN_OPTIMIZER_THEMES === '*') {
+    return ['v8light', 'v8dark', 'v7light', 'v7dark'];
+  }
```

Comment:
I think we may want to add some unit test on the now dynamic values of the registered uiSettings. Could just call `getThemeSettings` with the proper `dist` and `process.env.KBN_OPTIMIZER_THEMES` and assert on `result['theme:darkMode'].value` and so on.